### DISCUSSION TITLE
How to compress a  time series plot, after taking out specific month from a dataset?

### DISCUSSION THREAD

**Discussion by ghost** [pydata/xarray#5606] [Q&A] [answered]
Dear friends,

I have a dataset of OLR from 1986-2013 (daily data), and I am interested in plotting a time series which should have only boreal winter months i.e. from November to April.
(i) I am able to sort out Nov-Apr months from my datasets by using -
**OLRNA = OLR.sel(TIME = OLR.TIME.dt.month.isin([11,12,1,2,3,4]))**  and this is working.
(ii) But the problem is that whenever I am plotting a time series then that series is not continuous i.e not joining Nov-Apr for each year. I know that as I have selected only Nov-Apr months in my data so it's not showing. Then how to join or compress the time axis?

Kindly plz tell me how to plot this time series properly. I am attaching a plot also for the reference.
[Example.pdf](https://github.com/pydata/xarray/files/6820457/Example.pdf)

**Comment by dcherian** [comment, marked as answer]
The issue is that matplotlib is interpreting the time axis properly. If you really want different output I would drop the TIME variable: `OLRNA.drop_vars("TIME").plot()` You'll just have an integer x-axis.

**Reply by ghost** [reply]
I tried the same after some trials and got the proper plots. Basically what I did:-
After selecting specific month from the time series. Just plotted the series without considering 'time' on x-axis means just plotted the y-axis variable and let the x-axis denotes the serial numbers . And then with the help of matplotlib, just set the xticks and xticklabels manually wherever I want.
Thank you @dcherian as ur comment clarified me that I did correct.
